src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +268,37 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO recall(String trackingId, String centralCode, RecallDTO recallDTO) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var requestId = transaction.getHold().getFolioRequestId();
+    var request = requestService.findRequest(requestId);
+
+    try {
+      if (request.getStatus() == RequestDTO.RequestStatus.OPEN_AWAITING_PICKUP ||
+        request.getStatus() == RequestDTO.RequestStatus.OPEN_IN_TRANSIT) {
```

Comment:
add a variable for `request.getStatus()`, import for `RequestDTO.RequestStatus` and the expression will look much more short